api/v1alpha1/helpers.go

```diff
@@ -51,3 +53,17 @@ func (c *Contour) IsFinalized() bool {
 func (c *Contour) GatewayClassSet() bool {
 	return c.Spec.GatewayClassRef != nil
 }
+
+// MakeNodePorts returns a nodeport slice using the ports key as the nodeport name
+// and the ports value as the nodeport number.
+func MakeNodePorts(ports map[string]int) []NodePort {
```

Comment:
i would maybe argue for not putting this as a public function in the API just b/c its only used in tests right now

I think it is ok to have some duplication between packages, just I saw a lot of duplication in the e2e test file and wanted to cut down on that to make things easier to read